Fix ICE for user-defined value type in external library function (#16509)
ethereum/solidity: Changelog.md

* Yul EVM Code Transform: Improve stack shuffler performance by fixing a BFS deduplication issue.

Bugfixes:

> * Type System: Fix internal compiler error when using user-defined value types in storage arrays or mappings in external library functions.

Reviewer: Return parameters were unaffected since they are not a part of the signature, right?

```suggestion
* Type System: Fix internal compiler error when defining external library functions accepting storage pointers to arrays/mappings of user-defined value types.
```
